### PR TITLE
panel: Deindent code blocks when copying them to markdown.

### PR DESCRIPTION
**Pull request by emilio**
Otherwise copying heavily-indented code blocks ends up copying a lot of
whitespace.

### CODE REVIEW

**Comment by emilio** [comment]
@arai-a (couldn't mark you as a reviewer but if you want to take a look please do!)

File: static/js/panel.js [hunk 1]
```diff
@@ -218,21 +218,58 @@ var Panel = new (class Panel {
...
       const lineElem = document
         .getElementById(`line-${line}`)
         .querySelector(".source-line");
-      texts.push(lineElem.textContent.replace(/\n/, ""));
-
+      const lineText = lineElem.textContent.replace(/\n/, "");
+      commonWhitespacePrefix = computeCommonWhitespacePrefix(
+        lineText,
+        commonWhitespacePrefix
+      );
```
Comment (arai-a) [inline static/js/panel.js:259]: This doesn't deindent if there's empty line.
This code needs to be conditional on `lineText`.

File: static/js/panel.js [hunk 2]
```diff
@@ -218,21 +218,58 @@ var Panel = new (class Panel {
...
+      lines.push(lineText);
       lastLine = line;
     }
-    return texts;
+
+    if (commonWhitespacePrefix?.length) {
+      for (let i = 0; i < lines.length; ++i) {
+        if (lines[i] != kPlaceholder) {
+          lines[i] = lines[i].substring(commonWhitespacePrefix.length);
```
Comment (arai-a) [inline static/js/panel.js:267]: Once the empty line is excluded above, this should also be conditional on `lines[i]`

**Review by arai-a — COMMENTED** [review]
Thank you!
Added some comments.

File: static/js/panel.js [hunk 3]
```diff
@@ -203,19 +203,49 @@ var Panel = new (class Panel {
...
+        }
+      }
+
+      return lineText.substring(0, count);
+    }
+
     for (const line of [...Highlight.selectedLines].sort((a, b) => a - b)) {
       if (lastLine !== -1 && lastLine != line - 1) {
-        texts.push("...");
+        texts.push(kPlaceholder);
```
[the remark this answers is not in the document]
Comment (emilio) [reply static/js/panel.js]: Whops, nice catch.

File: static/js/panel.js (hunk 1 above)
Comment (emilio) [reply static/js/panel.js:264]: Made computeCommonWhitespacePrefix early-return instead.

File: static/js/panel.js (hunk 2 above)
Comment (emilio) [reply static/js/panel.js:272]: Done, thanks!

**Review by arai-a — APPROVED** [review]
Thanks!
looks good

**Pull request merged**